docs/compilation.rst

```diff
@@ -332,14 +332,35 @@ Optional CMake build parameters
 -  ``BUILD_NVTX`` - build with NVTX profiling enabled (default: OFF)
 -  ``BUILD_NVJPEG`` - build with ``nvJPEG`` support (default: ON)
 -  ``BUILD_LIBTIFF`` - build with ``libtiff`` support (default: ON)
+-  ``BUILD_FFMPEG`` - build with ``ffmpeg`` support (default: ON)
```

Comment:
I think it's redundant to have `BUILD_DALI_NODEPS` and `BUILD_OPENCV`+`BUILD_PROTOBUF`. I mean, the whole point of having `BUILD_DALI_NODEPS` is that we haven't `BUILD_OPENCV`+`BUILD_PROTOBUF` before, right? If we had them, it was possible to `BUILD_DALI_NODEPS` just by setting everything else off, right?

Reply:
Should we add to build scripts a proper support for:
- BUILD_DALI_KERNELS
- BUILD_DALI_PIPELINE
- BUILD_DALI_OPERATORS

Reply:
@szalpal Before the was no BUILD_OPENCV and BUILD_PROTOBUF, so I added them. The point of BUILD_DALI_NODEPS was to disable every dependency, including OpenCV and Protobuf. I do agree that there's little sense on having user facing options to disable OpenCV and Protobuf, since they are not very useful on their own because disabling them disables dali pipeline and operators as well. I suggest we downgrade those to just internal cmake variables that we set to off when BUILD_DALI_NODEPS=ON. Good point!

Reply:
@JanuszL BUILD_DALI_KERNELS BUILD_DALI_PIPELINE and BUILD_DALI_OPERATORS are just internal CMake variables, they are not exposed to the user as options, so we don't have to document or use them in the scripts in any way.

Reply:
ok